vscode/src/commands.ts

```diff
@@ -110,15 +118,267 @@ const commandsMap: (state: ExtensionState) => {
+            });
+            break;
+
+          case "ChatDeepSeek":
+            model = new ChatDeepSeek({
+              modelName: modelProvider.modelProvider.args["model"],
+              streaming: true,
+              temperature: modelProvider.modelProvider.args["temperature"] || 0,
+              maxTokens: modelProvider.modelProvider.args["max_tokens"],
+            });
+            break;
+
+          default:
+            throw new Error(`Unsupported model provider: ${providerType}`);
+        }
+
+        // need to grab this off the analysis results
+        const profile = getActiveProfile(state);
+        if (!profile) {
+          window.showErrorMessage("Profile undefined");
+          return;
+        }
+
+        // Prepare the system prompt
+        const systemPrompt = `
+You are an experienced java developer, who specializes in migrating code from ${profile.name}`;
+        console.log(systemPrompt);
+
+        // Process each file's incidents
+        const allDiffs: { original: string; modified: string; diff: string }[] = [];
+        const modifiedFiles = new Set<string>();
+
+        for (const [uri, fileIncidents] of Object.entries(incidentsByUri)) {
+          const parsedURI = Uri.parse(uri);
+          const relativePath = workspace.asRelativePath(parsedURI);
+
+          state.mutateData((draft) => {
+            draft.chatMessages.push(
+              {
+                messageToken: `m${Date.now()}`,
+                kind: ChatMessageType.String,
+                value: {
+                  message: `Analyzing file: ${relativePath} with incidents: ${(fileIncidents as EnhancedIncident[]).map((incident) => incident.violationId).join(", ")}\n\n`,
+                },
+                timestamp: new Date().toISOString(),
+              },
+              {
+                messageToken: `m${Date.now()}`,
+                kind: ChatMessageType.String,
+                value: { message: "" },
+                timestamp: new Date().toISOString(),
+              },
+            );
+          });
+
+          // Get the entire file content
+          const doc = await workspace.openTextDocument(parsedURI);
+          const fileContent = doc.getText();
+
+          // Prepare the incidents description for this file
+          const incidentsDescription = (fileIncidents as EnhancedIncident[])
+            .map((incident) => `* ${incident.lineNumber}: ${incident.message}`)
+            .join();
+
+          // Prepare the human prompt
+          const humanPrompt = `
+I will give you a file for which I want to take one step towards migrating ${profile.name}.
+I will provide you with static source code analysis information highlighting an issue which needs to be addressed.
+Fix all the issues described. Other problems will be solved in subsequent steps so it is unnecessary to handle them now.
+Before attempting to migrate the code ${profile.name} reason through what changes are required and why.
```

Comment:
/the code ${profile.name}/the code to ${profile.name}/

Reply:
I made it `the code from ${profile.name},` so it would be rendered: `the code from JavaEE to Quarkus`.